xapian-applications/omega/clickmodel/session.h

```diff
@@ -0,0 +1,57 @@
+/** @file session.h
+ * @brief Session class for handling search session data.
+ */
+/* Copyright (C) 2017 Vivek Pal
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <string>
+#include <vector>
+
+using namespace std;
```

Comment:
It's not good to dump everything from namespace `std` into the global namespace like this in a header which IIUC is intended for users to include in their own code.  Better to explicitly qualify with `std::` in a header (`using namespace std;` is OK for stuff that's internal to a program or library - the only real issue there is that a new C++ standard revision may result in a new symbol in `std` which collides with a symbol in the program, but a new C++ standard version seems to require a few tweaks anyway.)